apps/content-publishing-api/src/controllers/v3/content.controller.v3.ts

```diff
@@ -84,45 +109,82 @@ export class ContentControllerV3 {
         if (fileIndex === 0) {
           throw new BadRequestException('No files provided in the request');
         }
-
-        // Validate schema IDs match file count
         if (schemaIds.length !== fileProcessingPromises.length) {
           throw new BadRequestException('Number of schema IDs does not match number of files');
         }
 
-        // Process uploaded files
-        const uploadResults = await Promise.all(fileProcessingPromises);
+        const uploadResults = await Promise.allSettled(fileProcessingPromises);
+
+        const responseFiles: { referenceId?: string; cid?: string; error?: string }[] = [];
+        let hasFailedUploads = false;
+        const successfulUploads: { uploadResult: IFileResponse; originalIndex: number }[] = [];
+
+        uploadResults.forEach((uploadResult, index) => {
+          if (uploadResult.status === 'fulfilled' && uploadResult.value.cid && !uploadResult.value.error) {
+            successfulUploads.push({ uploadResult: uploadResult.value, originalIndex: index });
+            // Placeholder - will be filled after batch creation
+            responseFiles.push({ cid: uploadResult.value.cid });
+          } else {
+            hasFailedUploads = true;
+            const error =
+              uploadResult.status === 'rejected'
+                ? uploadResult.reason?.message || 'Upload failed'
+                : uploadResult.value.error || 'Upload failed';
+            responseFiles.push({ error });
+          }
+        });
```

Comment:
code golf; I like destructuring 😄 
just a suggestion; take it or leave it at your preference
```suggestion
        uploadResults.forEach(({ status, reason, value: uploadResult }, index) => {
          if (status === 'fulfilled' && !!uploadResult.cid && !uploadResult.error) {
            successfulUploads.push({ uploadResult, originalIndex: index });
            // Placeholder - will be filled after batch creation
            responseFiles.push({ cid: uploadResult.cid });
          } else {
            hasFailedUploads = true;
            responseFiles.push({ error: reason?.message || uploadResult?.error || 'Upload failed' });
          }
        });
```